src/features/courses/course-controller.ts

```diff
@@ -1148,6 +1150,73 @@ class CourseController {
 
         return result;
     };
+
+    async getEnrolledUserIdsInCourse({ courseId }: { courseId: number }): Promise<Array<number>> {
+        const course = await Course.findOne({
+            where: {
+                id: courseId
+            },
+            attributes: [],
+            include: [{
+                model: StudentEnrollment,
+                as: 'enrolledStudents',
+                attributes: ['userId'],
+            }]
+        });
+
+        if (_.isNil(course)) {
+            throw new RederlyError('getEnrolledUserIdsInCourse: Invalid course id');
+        }
+        const enrolledStudents = course.enrolledStudents;
+        if (_.isNil(enrolledStudents)) {
+            throw new RederlyError('getEnrolledUserIdsInCourse: course was missing enrolledStudents');
+        }
+        return enrolledStudents.map(enrolledStudent => enrolledStudent.userId);
+    }
+
+    async getEnrolledUserIdsInTopic({ topicId }: { topicId: number }): Promise<Array<number>> {
+        const topics = await CourseTopicContent.findAll({
+            where: {
+                id: topicId
+            },
+            attributes: [],
+            include: [{
+                model: CourseUnitContent,
+                as: 'unit',
+                attributes: ['courseId'],
+                include:[{
+                    model: Course,
+                    as: 'course',
+                    attributes: ['id'],
+                    include: [{
+                        model: StudentEnrollment,
+                        as: 'enrolledStudents',
+                        attributes: ['userId'],
+                    }]
+                }]
+            }]
+        });
+
+        if (topics.length === 0) {
+            throw new RederlyError('getEnrolledUserIdsInTopic: fetch for topics had 0 results');
+        }
+        if (topics.length > 1) {
+            logger.warn('getEnrolledUserIdsInTopic fetched more than 1 topic, expected only one');
+        }
+        const unit = topics[0].unit;
```

Comment:
Do we have `.first` on the backend?

Reply:
We do have .first, however, I was already checking length anyway so this is null safe (first would have made me check it again, I guess I could only check first in the 0 check earlier (probably would have but I refactored the code a bit and this isn't how it was initally organized)